File: Symfony/src/BGR/Serrano/ProductoBundle/Entity/Presentacion.php
<?php

namespace BGR\Serrano\ProductoBundle\Entity;

use Doctrine\ORM\Mapping as ORM;
use JMS\Serializer\Annotation\Type;
use JMS\Serializer\Annotation\Exclude;

/**
 * Presentacion
 *
 * @ORM\Table()
 * @ORM\Entity(repositoryClass="BGR\Serrano\ProductoBundle\Entity\PresentacionRepository")
 */
class Presentacion
{
    /**
     * @var integer
     *
     * @ORM\Column(name="id", type="integer")
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     * @Type("integer")
     */
    private $id;
 
     /**
     * @var string
     *
     * @ORM\Column(name="descripcion", type="string", length=255)
     * @Type("string")
     */
    private $descripcion;

    /**
     * @var boolean
     *
     * @ORM\Column(name="active", type="boolean")
     * @Type("boolean")
     */
    private $active;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="date_create", type="datetime")
     * @Type("DateTime<'Y-m-d'>")
     */
    private $date_create;

    /**
     * @var \DateTime
     *
     * @ORM\Column(name="date_update", type="datetime")
     * @Type("DateTime<'Y-m-d'>")
     */
    private $date_update;

    /**
     * @var boolean
     *
     * @ORM\Column(name="fraccionable", type="boolean")
     * @Type("boolean")
     */
    private $fraccionable;

    /**
     * @var float
     *
     * @ORM\Column(name="peso_neto", type="decimal", length=255)
     * @Type("double")
     */
    private $peso_neto;

    /**
     * @var float
     *
     * @ORM\Column(name="peso_escurrido", type="decimal", length=255)
     * @Type("double")
     */
    private $peso_escurrido;

    /**
     * @var integer
     *
     * @Type("integer")
     */
    protected $cantidad_paquetes;
    
    /**
     * @var integer
     * @ORM\Column(name="cantidad", type="decimal", length=255)
     * @Type("integer")
     */
    protected $cantidad;

    /**
     * Get id
     *
     * @return string 
     */
    public function getId()
    {
        return $this->id;
    }

    /**
     * Set active
     *
     * @param boolean $active
     * @return Presentacion
     */
    public function setActive($active)
    {
        $this->active = $active;
    
        return $this;
    }

    /**
     * Get active
     *
     * @return boolean 
     */
    public function getActive()
    {
        return $this->active;
    }

    /**
     * Set date_create
     *
     * @param \DateTime $date_create
     * @return Presentacion
     */
    public function setDate_create($date_create)
    {
        $this->date_create = $date_create;
    
        return $this;
    }

    /**
     * Get date_create
     *
     * @return \DateTime 
     */
    public function getPeso_neto()
    {
        return $this->peso_neto;
    }

     /**
     * Set peso_neto
     *
     * @param \float $peso_neto
     * @return Presentacion
     */
    public function setPeso_neto($peso_neto)
    {
        $this->peso_neto = $peso_neto;
    
        return $this;
    }


    /**
     * Get peso_escurrido
     *
     * @return \DateTime 
     */
    public function getPeso_escurrido()
    {
        return $this->peso_escurrido;
    }
     /**
     * Set peso_escurrido
     *
     * @param \float $peso_escurrido
     * @return Presentacion
     */
    public function setPeso_escurrido($peso_escurrido)
    {
        $this->peso_escurrido = $peso_escurrido;
    
        return $this;
    }
    /**
     * Get date_create
     *
     * @return \DateTime 
     */
    public function getDate_create()
    {
        return $this->date_create;
    }

    /**
     * Set dateUpdate
     *
     * @param \DateTime $dateUpdate
     * @return Presentacion
     */
    public function setDate_update($date_update)
    {
        $this->date_update = $date_update;
    
        return $this;
    }

    /**
     * Get date_update
     *
     * @return \DateTime 
     */
    public function getDate_update()
    {
        return $this->date_update;
    }

    /**
     * Set fraccionable
     *
     * @param string $fraccionable
     * @return Presentacion
     */
    public function setFraccionable($fraccionable)
    {
        $this->fraccionable = $fraccionable;
    
        return $this;
    }

    /**
     * Get fraccionable
     *
     * @return string 
     */
    public function getFraccionable()
    {
        return $this->fraccionable;
    }

    /**
     * @ORM\ManyToOne(targetEntity="Material") 
     * @ORM\JoinColumn(name="material_id", referencedColumnName="id")
     * @Type("BGR\Serrano\ProductoBundle\Entity\Material")
     */
    protected $material;

    public function setMaterial(Material $material)
    {
        $this->material = $material;
    }

    public function getMaterial()
    {
        return $this->material;
    }


    /**
     * @ORM\ManyToOne(targetEntity="Lote") 
     * @ORM\JoinColumn(name="lote_id", referencedColumnName="id")
     * @Type("BGR\Serrano\ProductoBundle\Entity\Lote")
     */
    protected $lote;

    public function setLote(Lote $lote)
    {
        $this->lote = $lote;
    }

    public function getLote()
    {
        return $this->lote;
    }   
   /**
     * @ORM\ManyToOne(targetEntity="UnidadDeMedida") 
     * @ORM\JoinColumn(name="unidadDeMedida_id", referencedColumnName="id")
     * @Type("BGR\Serrano\ProductoBundle\Entity\UnidadDeMedida")
     */
    protected $unidad_de_medida;

    public function setUnidad_de_medida(UnidadDeMedida $unidad_de_medida)
    {
        $this->unidad_de_medida = $unidad_de_medida;
    }

    public function getUnidad_de_medida()
    {
        return $this->unidad_de_medida;
    }
    /**
     * @ORM\ManyToOne(targetEntity="Producto") 
     * @ORM\JoinColumn(name="producto_id", referencedColumnName="id")
     * @Type("BGR\Serrano\ProductoBundle\Entity\Producto")
     *
     */
    protected $producto;

    public function setProducto(Producto $producto)
    {
        $this->producto = $producto;
    }

    public function getProducto()
    {
        return $this->producto;
    }
	public function getCantidadPaquetes() {
		return $this->cantidad_paquetes;
	}
	public function setCantidadPaquetes($cantidad_paquetes) {
		$this->cantidad_paquetes = $cantidad_paquetes;
		return $this;
	}
	
	public function getCantidad() {
		return $this->cantidad;
	}
	public function setCantidad($cantidad) {
		$this->cantidad = $cantidad;
		return $this;
	}
	

}

File: BGR/BGR-Data/src/main/java/ar/com/bgr/serrano/model/Presentacion.java
package ar.com.bgr.serrano.model;

import java.util.Date;
import java.util.Set;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.Table;

import org.hibernate.annotations.IndexColumn;

/**
 * 
 * @author matias
 *
 */
@Entity(name="Presentacion")
@Table(name="Presentacion")
public class Presentacion {
	
	@Id
	@GeneratedValue
	private int id;
	
	@Column(name="descripcion")
	private String descripcion;	
	
	@Column(name="active")
	private Boolean active;

	@Column(name="date_create")
	private Date date_create;

	@Column(name="date_update")
	private Date date_update;

	@Column(name="fraccionable")
	private Boolean fraccionable;

	@Column(name="peso_neto")
    private Double peso_neto;
	
	@Column(name="peso_escurrido")
    private Double peso_escurrido;

	@Column(name="cantidad_paquetes")
    private Integer cantidad_paquetes;
    
	@Column(name="cantidad")
    private Double cantidad; 
    
    @Column(name="message")
    private String message;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name="material_id")
    private Material material;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name="proveedor_id")
    private Proveedor proveedor;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name="lote_id")
    private Lote lote;

    @ManyToOne
    @JoinColumn(name="unidadDeMedida_id")
    private UnidadDeMedida unidad_de_medida;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name="producto_id")
    private Producto producto;
	
	@OneToMany(fetch = FetchType.EAGER, cascade={CascadeType.ALL})
    @JoinColumn(name="presentacion_id")
    @IndexColumn(name="id")
    private Set<Paquete> paquetes;

    public Presentacion(){}
    
	/**
	 * @return the id
	 */
	public int getId() {
		return id;
	}


	/**
	 * @param id the id to set
	 */
	public void setId(int id) {
		this.id = id;
	}


	/**
	 * @return the descripcion
	 */
	public String getDescripcion() {
		return descripcion;
	}


	/**
	 * @param descripcion the descripcion to set
	 */
	public void setDescripcion(String descripcion) {
		this.descripcion = descripcion;
	}


	/**
	 * @return the active
	 */
	public Boolean getActive() {
		return active;
	}


	/**
	 * @param active the active to set
	 */
	public void setActive(Boolean active) {
		this.active = active;
	}


	/**
	 * @return the date_create
	 */
	public Date getDate_create() {
		return date_create;
	}


	/**
	 * @param date_create the date_create to set
	 */
	public void setDate_create(Date date_create) {
		this.date_create = date_create;
	}


	/**
	 * @return the date_update
	 */
	public Date getDate_update() {
		return date_update;
	}


	/**
	 * @param date_update the date_update to set
	 */
	public void setDate_update(Date date_update) {
		this.date_update = date_update;
	}


	/**
	 * @return the fraccionable
	 */
	public Boolean getFraccionable() {
		return fraccionable;
	}


	/**
	 * @param fraccionable the fraccionable to set
	 */
	public void setFraccionable(Boolean fraccionable) {
		this.fraccionable = fraccionable;
	}


	/**
	 * @return the peso_neto
	 */
	public Double getPeso_neto() {
		return peso_neto;
	}


	/**
	 * @param peso_neto the peso_neto to set
	 */
	public void setPeso_neto(Double peso_neto) {
		this.peso_neto = peso_neto;
	}


	/**
	 * @return the peso_escurrido
	 */
	public Double getPeso_escurrido() {
		return peso_escurrido;
	}


	/**
	 * @param peso_escurrido the peso_escurrido to set
	 */
	public void setPeso_escurrido(Double peso_escurrido) {
		this.peso_escurrido = peso_escurrido;
	}


	/**
	 * @return the cantidad_paquetes
	 */
	public Integer getCantidad_paquetes() {
		return cantidad_paquetes;
	}


	/**
	 * @param cantidad_paquetes the cantidad_paquetes to set
	 */
	public void setCantidad_paquetes(Integer cantidad_paquetes) {
		this.cantidad_paquetes = cantidad_paquetes;
	}


	/**
	 * @return the cantidad
	 */
	public Double getCantidad() {
		return cantidad;
	}


	/**
	 * @param cantidad the cantidad to set
	 */
	public void setCantidad(Double cantidad) {
		this.cantidad = cantidad;
	}


	/**
	 * @return the message
	 */
	public String getMessage() {
		return message;
	}


	/**
	 * @param message the message to set
	 */
	public void setMessage(String message) {
		this.message = message;
	}


	/**
	 * @return the material
	 */
	public Material getMaterial() {
		return material;
	}


	/**
	 * @param material the material to set
	 */
	public void setMaterial(Material material) {
		this.material = material;
	}


	/**
	 * @return the proveedor
	 */
	public Proveedor getProveedor() {
		return proveedor;
	}


	/**
	 * @param proveedor the proveedor to set
	 */
	public void setProveedor(Proveedor proveedor) {
		this.proveedor = proveedor;
	}


	/**
	 * @return the lote
	 */
	public Lote getLote() {
		return lote;
	}


	/**
	 * @param lote the lote to set
	 */
	public void setLote(Lote lote) {
		this.lote = lote;
	}


	/**
	 * @return the unidad_de_medida
	 */
	public UnidadDeMedida getUnidad_de_medida() {
		return unidad_de_medida;
	}


	/**
	 * @param unidad_de_medida the unidad_de_medida to set
	 */
	public void setUnidad_de_medida(UnidadDeMedida unidad_de_medida) {
		this.unidad_de_medida = unidad_de_medida;
	}


	/**
	 * @return the producto
	 */
	public Producto getProducto() {
		return producto;
	}


	/**
	 * @param producto the producto to set
	 */
	public void setProducto(Producto producto) {
		this.producto = producto;
	}
    

    /**
	 * @return the paquete
	 */
	public Set<Paquete> getPaquetes() {
		return paquetes;
	}

	/**
	 * @param paquete the paquete to set
	 */
	public void setPaquetes(Set<Paquete> paquetes) {
		this.paquetes = paquetes;
	}
    
    
    
}
